spec.md

```diff
@@ -0,0 +1,954 @@
+
+Identity service RPCs allow a CMI-Client to query a CMI-Plugin for capabilities, health, and other metadata.
+The general flow of the success case MAY be as follows (protos illustrated in YAML for brevity):
+
+1. CMI-Client queries metadata via Identity RPC.
+
+```
+   # CMI-Client --(GetPluginInfo)--> Plugin
+   request:
+   response:
+      name: org.foo.whizbang.super-plugin
+      vendor_version: blue-green
+      manifest:
+        baz: qaz
+```
+
+2. CMI-Client queries available capabilities of the plugin.
+
+```
+   # CMI-Client --(GetPluginCapabilities)--> Plugin
+   request:
+   response:
+     capabilities:
+       - service:
+           type: CONTROLLER_SERVICE
+```
+
+3. CMI-Client queries the readiness of the plugin.
+
+```
+   # CMI-Client --(Probe)--> Plugin
+   request:
+   response: {}
+```
+
+#### `GetPluginInfo`
+
+```protobuf
+message GetPluginInfoRequest {
+    // Intentionally empty.
+}
+
+message GetPluginInfoResponse {
+    // The name MUST follow domain name notation format
+    // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
+    // include the plugin's host company name and the plugin name,
+    // to minimize the possibility of collisions. It MUST be 63
+    // characters or less, beginning and ending with an alphanumeric
+    // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
+    // alphanumerics between.
+    // This field is REQUIRED.
+    string name = 1;
+
+    // The version specifies the plugin version
+    // Value of this field is opaque to the CMI-Client.
+    // This field is REQUIRED.
+    string version = 2;
+
+    // manifest contains a map of key-value pairs to pass
+    // any additonal information about the plugin.
+    // Values are opaque to the CMI-Client.
+    // This field is OPTIONAL.
+    map<string, string> manifest = 3;
+}
+```
+
+##### GetPluginInfo Errors
+
+If the plugin is unable to complete the GetPluginInfo call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+
+#### `GetPluginCapabilities`
+
+This REQUIRED RPC allows the CMI-Client to query the supported capabilities of the Plugin "as a whole": it is the grand sum of all capabilities.
+All CMI-Plugins of the same version (see `version` of `GetPluginInfoResponse`) of the Plugin SHALL return the same set of capabilities.
+
+```protobuf
+
+message GetPluginCapabilitiesRequest {
+    // Intentionally empty.
+}
+
+message GetPluginCapabilitiesResponse {
+    // All the capabilities that the controller service supports.
+    // This field is OPTIONAL.
+    repeated PluginCapability capabilities = 1;
+}
+
+// Specifies a capability of the CMI-Plugin.
+message PluginCapability {
+    message RPC {
+        enum Type {
+            // UNKNOWN is used to specific an capaability beyond the set
+            // provided below
+            UNKNOWN = 0;
+
+            // CREATE_MACHINE tells that the plugin implements the
+            // CreateMachine() interface
+            CREATE_MACHINE = 1;
```

Comment:
As mentioned above, I would remove `CREATE_MACHINE` and `DELETE_MACHINE` and treat them as core capabilities that **must** be implemented. Also, I don't see a capability for `LIST_MACHINES` - is this intentional? I believe this call should be treated as non-core and therefore a capability for it should exist (but I may be wrong).